WARNING:django.request:Not Found: /api/device/
WARNING:django.request:Not Found: /favicon.ico
DEBUG:MARKDOWN:Successfully imported extension module "markdown.extensions.toc".
DEBUG:MARKDOWN:Successfully loaded extension "markdown.extensions.toc.TocExtension".
WARNING:django.request:Method Not Allowed: /api/device/delete/Openwrt-VM/
WARNING:django.request:Not Found: /favicon.ico
ERROR:django.request:Internal Server Error: /api/device/delete/Openwrt-VM/
Traceback (most recent call last):
  File "/opt/vi/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/opt/vi/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/opt/vi/.venv/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/rest_framework/decorators.py", line 50, in handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/server/api/views.py", line 82, in removeDevice
    device = DeviceHandler.objects.get(devicename=dn)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/vi/.venv/lib/python3.11/site-packages/django/db/models/query.py", line 649, in get
    raise self.model.DoesNotExist(
api.models.DeviceHandler.DoesNotExist: DeviceHandler matching query does not exist.
WARNING:django.request:Not Found: /api/device/get/Openwrt-VM/
WARNING:django.request:Not Found: /favicon.ico
WARNING:django.request:Not Found: /api/device/get/Openwrt-VM/
WARNING:django.request:Not Found: /api/device/get/OpenWRT-VM/
WARNING:django.request:Not Found: /favicon.ico
